src/task.ts

```diff
@@ -0,0 +1,450 @@
+import { RetryOptions } from ".";
+import { IAction, CreateTimerAction } from "./classes";
+import { TaskOrchestrationExecutor } from "./taskorchestrationexecutor";
+
+/**
+ * @hidden
+ * The states a task can be in
+ */
+export enum TaskState {
+    Running,
+    Failed,
+    Completed,
+}
+
+/**
+ * @hidden
+ * A taskID, either a `string` for external events,
+ * or either `false` or a `number` for un-awaited
+ * an awaited tasks respectively.
+ */
+export type TaskID = number | string | false;
+
+/**
+ * @hidden
+ * A backing action, either a proper action or "noOp" for an internal-only task
+ */
+export type BackingAction = IAction | "noOp";
+
+/**
+ * A Durable Functions Task.
+ */
+export interface ITask {
+    /**
+     * Whether the task has completed. Note that completion is not
+     * equivalent to success.
+     */
+    isCompleted: boolean;
+    /**
+     * Whether the task faulted in some way due to error.
+     */
+    isFaulted: boolean;
+    /**
+     * The result of the task, if completed. Otherwise `undefined`.
+     */
+    result?: unknown;
+}
+
+/**
+ * Returned from [[DurableOrchestrationClient]].[[createTimer]] if the call is
+ * not `yield`-ed. Represents a pending timer. See documentation on [[Task]]
+ * for more information.
+ *
+ * All pending timers must be completed or canceled for an orchestration to
+ * complete.
+ *
+ * @example Cancel a timer
+ * ```javascript
+ * // calculate expiration date
+ * const timeoutTask = context.df.createTimer(expirationDate);
+ *
+ * // do some work
+ *
+ * if (!timeoutTask.isCompleted) {
+ *     timeoutTask.cancel();
```

Comment:
In my previous review, I suggested adding a comment explaining the importance of calling `cancel()` on the task and got a thumbs up. Just checking to see if you were still planning on doing that. Looks like there's another example below too.

Reply:
I had meant to add this comment. My next commits adds it.